code/modules/projectiles/guns/ballistic.dm

```diff
@@ -398,7 +400,7 @@
 		user.visible_message("<span class='danger'>\The [src] goes off!</span>", "<span class='danger'>\The [src] goes off in your face!</span>")
 		return
 
-	if(do_after(user, 30, target = src))
+	if(do_after(user, max(30, 60 - I.force), target = src))
```

Comment:
this equation feels a bit wonnky; the force of the weapon barely affects the time that it takes to saw off a shotgun (it only lowers it by a decisecond per point of force).